app/Jobs/CheckForUpdatesJob.php

```
@@ -22,20 +23,60 @@ public function handle(): void
+                        'attempted_best' => $bestVersion,
+                        'using' => $current_version,
+                    ]);
+                    $bestVersion = $current_version;
+                }
+
+                // Use data_set() for safe mutation (fixes #3)
+                data_set($versions, 'coolify.v4.version', $bestVersion);
+                $latest_version = $bestVersion;
+
+                // ALWAYS write versions.json (for Sentinel, Helper, Traefik updates)
+                File::put(base_path('versions.json'), json_encode($versions, JSON_PRETTY_PRINT));
```

PHPMD (2.15.0)

Avoid using static access to class '\Illuminate\Support\Facades\File' in method 'handle'. (undefined)

(StaticAccess)